src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -117,7 +103,7 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
 
     // run job
     val repartitionedIndexDataFrame =
-      indexDataFrame.repartition(numBuckets, resolvedIndexedColumns.map(df(_)): _*)
+      indexDataFrame.repartition(numBuckets, resolvedIndexedColumns.map(c => col(s"$c")): _*)
```

Comment:
can you explain why this change was required?

Reply:
The `prepareIndexDataFrame` does the things a bit different. First, from the original data frame it has to extract the nested field values to the top level. Second, it gives this new top level field an escaped name so that we can save it properly and retrieve it safely.

So, `indexDataFrame` columns are different from the original `df` data frame and doing `df(_)` with an unknown columns would result it exceptions.

The `resolvedIndexColumns` contains escaped field names and the `indexDataFrame` contains those names.